syntax/parser.go

```diff
@@ -419,6 +419,8 @@ func (p *parser) parseForStmt() Stmt {
 	var initializer Stmt
 	if p.tok == LET {
 		initializer = p.parseLetStmt()
+	} else if p.tok != SEMICOLON {
+		initializer = p.parseStmt()
```

Comment:
In this case,
```go
if p.tok == LET
```
branch could be removed.

However, `p.parseStmt` parses all statements which allows people to write:

```js
for (return;;) {}
```

or even

```js
for (for ({};) {};) {}
```

That is a bit weird.

Maybe we could follow the JavaScript's syntax?

Reply:
Replace `p.parseStmt` with `p.parseSimpleStmt` is a simpler choice.